plans/query_plans.go

```diff
@@ -708,11 +727,32 @@ func (plan *QueryPlan) Select() ([]interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
+	if plan.cacheable {
+		cacheKey := fmt.Sprintf("%s: %v", query, plan.args)
+		data, err := getCacheData(cacheKey, plan.memCache)
+		if err == nil { // fail silently - graceful fallback
+			return data, nil
```

Comment:
`data` needs to be unmarshaled to a slice of the same type as `plan.target`, here.  Otherwise, the query is going to return a different type depending on whether or not the data is in cache.